plugin/plugin_test.go

```diff
@@ -0,0 +1,127 @@
+package plugin
+
+import (
+	context "context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAeadPluginWrapper(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+
+	pluginPath := os.Getenv("PLUGIN_PATH")
+	if pluginPath == "" {
+		t.Skipf("skipping plugin test as no PLUGIN_PATH specified")
+	}
+
+	wrapper, cleanup := TestPlugin(t, pluginPath)
+
+	require.NotNil(cleanup)
+	defer cleanup()
```

Comment:
Looks like you do cleanup via `t.Cleanup` in other tests, should we also do it that way here?

Reply:
Sure.